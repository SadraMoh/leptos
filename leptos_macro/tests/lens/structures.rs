use leptos_macro::Lens;

#[derive(Lens)]
enum FailOnEnum {
    This(u32),
    Should(String),
    Panic,
}

#[derive(Lens)]
union FaulOnUnion {
    this: u32,
    should: std::mem::ManuallyDrop<String>,
    panic: std::mem::ManuallyDrop<Box<String>>,
}

fn main() {}
