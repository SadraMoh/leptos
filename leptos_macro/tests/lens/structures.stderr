error: proc-macro derive panicked
 --> tests/lens/structures.rs:3:10
  |
3 | #[derive(Lens)]
  |          ^^^^
  |
  = help: message: Lens cannot be derived by enums or unions

error: proc-macro derive panicked
  --> tests/lens/structures.rs:10:10
   |
10 | #[derive(Lens)]
   |          ^^^^
   |
   = help: message: Lens cannot be derived by enums or unions
